src/scmrepo/git/backend/dulwich/__init__.py

```diff
@@ -580,7 +580,7 @@ def update_refs(refs):
             result = client.send_pack(
                 path,
                 update_refs,
-                self.repo.object_store.generate_pack_data,
+                generate_pack_data=self.repo.generate_pack_data,
```

Comment:
`repo.generate_pack_data` wraps `repo.object_store.generate_pack_data`, but handling for shallow commits is only done in the `repo` level wrapper